node/core/approval-voting/src/lib.rs

```diff
@@ -774,6 +776,159 @@ where
 	Ok(())
 }
 
+#[derive(Eq, PartialEq)]
+enum RecoveryState {
+	Pending,
+	Complete,
+}
+
+impl RecoveryState {
+	fn complete(&mut self) -> bool {
+		let complete = *self == RecoveryState::Complete;
+		*self = RecoveryState::Complete;
+		complete
+	}
+}
+
+// Determines the set of assigned but unapproved candidates that require resumption of the
+// approval process by the local node. For each such, this method generates a LaunchApproval
+// action so that the approvals can continue to make progress after a restar.
+async fn handle_startup(
+	overlay_db: &mut OverlayedBackend<'_, impl Backend>,
+	state: &mut State,
+) -> SubsystemResult<Vec<Action>> {
+	let all_candidates = overlay_db.load_all_candidates().map_err(|e| {
+		tracing::error!(target: LOG_TARGET, "Failed initial load of active candidates: {:?}", e);
+		e
+	})?;
+
+	let candidates_by_session: HashMap<_, _> =
+		all_candidates
+			.into_iter()
+			.fold(HashMap::new(), |mut candidates_by_session, candidate| {
+				candidates_by_session
+					.entry(candidate.session)
+					.or_insert_with(Vec::new)
+					.push(candidate);
+				candidates_by_session
+			});
+
+	let mut actions = Vec::new();
+	for (session, candidates) in candidates_by_session {
+		let info = match state.session_info(session) {
+			Some(info) => info,
+			None => {
+				tracing::warn!(target: LOG_TARGET, "Missing session info for {:?}", session);
```

Comment:
I think it would be normal if the candidates we have are from an old session (e.g. the node was off for a while). I would suggest downgrading this to `info!` or even `debug!` to not let people report issues about this warning.